src/main/java/com/gorlah/kappabot/integration/discord/DiscordBot.java

```diff
@@ -19,74 +20,75 @@
 import java.util.Map;
 
 @Component
+@RequiredArgsConstructor
 public class DiscordBot extends ListenerAdapter {
-    
-    @Value("${discord.token}")
-    private String discordToken;
-    
+
     @Value("${discord.command.prefix}")
     private String commandPrefix;
-    
-    private JDA bot;
-    
+
     private final CommandProcessor commandProcessor;
-    
-    public DiscordBot(CommandProcessor commandProcessor) {
-        this.commandProcessor = commandProcessor;
-    }
-    
+    private final DiscordIntegration discordIntegration;
+
+    private JDA bot;
+
     @PostConstruct
-    private void initialize() throws LoginException {
-        bot = new JDABuilder(discordToken)
-                .addEventListeners(this)
-                .setActivity(Activity.playing(commandPrefix + " help | I'm back. \uD83D\uDE0F"))
-                .build();
+    private void initialize()
+            throws LoginException {
+        if (discordIntegration.isEnabled()) {
+            bot = new JDABuilder(discordIntegration.getToken())
+                    .addEventListeners(this)
+                    .setActivity(Activity.playing(commandPrefix + " help | I'm back. \uD83D\uDE0F"))
+                    .build();
+        }
     }
-    
+
     @Override
     public void onMessageReceived(MessageReceivedEvent event) {
         if (event.getAuthor().isBot()) {
             return;
         }
-        
+
         String message = event.getMessage().getContentRaw();
         StringBuilder response = new StringBuilder();
-        
+
         if (SlashdotParser.find(message)) {
             ArrayList<String> titles = SlashdotParser.getTitles(message);
-    
+
             for (int i = 0; i < titles.size(); i++) {
                 if (i > 0) {
                     response.append("\n");
                 }
-                
+
                 response.append("> ").append(titles.get(i));
             }
         } else {
             if (message.length() < commandPrefix.length()
-                    || !commandPrefix.equalsIgnoreCase(message.substring(0, commandPrefix.length()))) {
+                    || !commandPrefix.equalsIgnoreCase(message.substring(
+                    0,
+                    commandPrefix.length()))) {
```

Comment:
Was this code styling intended?

Reply:
That's just how the code style rules decided to chop it down, I will reformat it in a more pleasing way.

Reply:
Okay so I guess the code style needs to be updated then.  I can look into that.  Thanks for fixing it.